apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -545,23 +588,28 @@ impl std::fmt::Display for CachingQueryKey {
         hasher.update(
             &serde_json::to_vec(&self.plan_options).expect("serialization should not fail"),
         );
+        hasher
+            .update(&serde_json::to_vec(&self.config_mode).expect("serialization should not fail"));
         hasher.update(&serde_json::to_vec(&self.sdl).expect("serialization should not fail"));
+        hasher.update([self.introspection as u8]);
```

Comment:
why is introspection added here but not in the Hash implementation below?

Reply:
Nice catch: [43ccd9d](https://github.com/apollographql/router/pull/5100/commits/43ccd9d21b7d6db5e4c16d9d6a2107169e202d57)

I am wondering though why we are using a manual implementation of `Hash` rather than using derive? There seems to be nothing special about this structure that would warrant a custom impl.

Reply:
I'd have to dig it up, but I think there was something in the key that could not implement `Hash` in the past

Reply:
Once we figure that out, is there somewhere we could add a comment to make that more intuitive?

Reply:
If we don't need a custom implementation now let's remove it and use derive. That way we don't have to worry about it in the future. If we do eventually need a custom version would need to comment and possibly fuzz test.